daffodil-runtime2/src/main/resources/org/apache/daffodil/runtime2/c/libcli/daffodil_getopt.c

```diff
@@ -17,10 +17,10 @@
 
 // clang-format off
 #include "daffodil_getopt.h"
-#include <string.h>      // for strcmp, strrchr
-#include <unistd.h>      // for optarg, getopt, optopt, optind
-#include "cli_errors.h"  // for CLI_UNEXPECTED_ARGUMENT, CLI_HELP_USAGE, CLI_INVALID_COMMAND, CLI_INVALID_INFOSET, CLI_INVALID_OPTION, CLI_MISSING_COMMAND, CLI_MISSING_VALUE, CLI_PROGRAM_ERROR, CLI_PROGRAM_VERSION
-#include "version.h"
+#include <string.h>            // for strcmp, strrchr
+#include <unistd.h>            // for optarg, getopt, optopt, optind
+#include "cli_errors.h"        // for CLI_UNEXPECTED_ARGUMENT, CLI_HELP_USAGE, CLI_INVALID_COMMAND, CLI_INVALID_INFOSET, CLI_INVALID_OPTION, CLI_MISSING_COMMAND, CLI_MISSING_VALUE, CLI_PROGRAM_ERROR, CLI_PROGRAM_VERSION
+#include "daffodil_version.h"  // for daffodil_program_version
```

Comment:
`CodeGenerator.scala` also needs a change to reference `daffodil_version.h` instead of `version.h`, since it overwrites the file with the correct version number.

Though, another option might be to create an sbt resource generator that generates `daffodil_version.h` rather than having a dummy file that the code generator overwrites.  I can see if I can whip that up if you like that better.

Reply:
I'd made/tested my changes in a generated output directory and I'd forgotten about CodeGenerator.scala, sigh.  

Your sbt resource generator sounds good, but I'll go ahead and update CodeGenerator.scala since that'll be quicker.  Total number of lines of code probably will be smaller this way too.

Reply:
Fair enough. Another thought, should the `ifndef`/`define` in the file be changed from `VERSION_H` to `DAFFODIL_VERSION_H`?